### PR TITLE
Journalbeat matches support && minor additions

### PR DESCRIPTION
**Pull request by kvch**
### Matching support
From now on it's possible to match for journal entry fields in Journalbeat using the new option `matches`. This requires a list of key value pairs separated by "=". The key has to be a journalbeat event key (e.g systemd.unit) and the value is the exact value journal reader needs to find in the entries.

Example configuration which returns NGINX and dhclient entries from the journal:
```yml
include_matches:
  - "systemd.unit=nginx"
  - "process.name=dhclient"
```
### Docker fields
Added docker fields from: https://docs.docker.com/config/containers/logging/journald/
- `container.id`
- `container.id_truncated`
- `container.name`
- `container.image.tag`
- `container.partial`

### Parse timestamp of entries
Journalbeat parses the timestamp of the entry and adds it to the event as `@timestamp`. The time of reading by Journalbeat is saved in `read_timestamp`.

### Save custom fields
Custom fields by various sources are stored under `custom`. Field names are normalized, meaning `"_"` prefix is removed and every letter is lowercase.

### Fields && processors
From now on it is possible to configure `processors` and `fields`, etc on `input` level.

### Metrics

The size of each open reader is reporting in bytes:
```
{
  "journalbeat": {
    "journals": {
      "journal_1": {
         "path": "system.journal",
         "size_in_bytes": 123124214,
       }
    }
}
```

### CODE REVIEW

**Comment by ph** [comment]
@kvch I will wait for the other to get merged, I don't know whats new :)

**Comment by kvch** [comment]
@ph I am waiting for you to approve my other PR, so I can merge it ;)

File: journalbeat/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM golang:1.10.3
```
Comment (ruflin) [inline journalbeat/Dockerfile]: I think this docker file will become useful but I wonder if it is used in the tests or if it was missing before?

Comment (kvch) [reply journalbeat/Dockerfile]: It was missing before. I has every dependency installed, so everyone can build and run `journalbeat`.

**Comment by kvch** [comment]
I renamed `matches` to `include_matches`. It's much better name, thanks for the suggestion.
One more reason why I went with `"systemd.unit=nginx"` is that Journalbeat does not parse these expressions. It simply passes it to the journal reader which does the matching for us.

Processors are different, because the Beat does the processing for you, not an outside lib. So I don't think it's required to configure `include_matches` the same as processors. However, if it leads to a better cleaner configuration I am all for changing it.

**Comment by ruflin** [comment]
I'm good with keeping these config options. I would probably rather change the other ones to make them shorter :-)

**Comment by ph** [comment]
@kvch Changes LGTM, I will test it today with the filtering.

**Comment by ph** [comment]
@kvch Also can you address Hound's comments?

**Comment by kvch** [comment]
@ph I have added comments, but it does not seem to be satisfied: https://github.com/elastic/beats/pull/8324/commits/ad7ddcda58f2229d16b15247f0201646265c2cc9

**Comment by kvch** [comment]
Failing tests are unrelated.

**Pull request merged**